tests/crypto_test.py

```diff
@@ -274,8 +307,12 @@ def test_signed_only_false(self):
 
     @staticmethod
     def _context_mock():
-        error = gpgme.GpgmeError()
-        error.code = gpgme.ERR_AMBIGUOUS_NAME
+        # We need an uninittialized class that is a descendent of GPGMEError
+        # for the test, but we need to make the getcode method a mock, so we
+        # use type to make a dynamic sublcass, since raise wont' accept
+        # anything that isn't an Exception
+        error = type('CustomError', (gpg.errors.GPGMEError, ), {})
+        error.getcode = mock.Mock(return_value=gpg.errors.AMBIGUOUS_NAME)
```

Comment:
No, I doubt many people know that signature for `type`. `type` is a bit of a mysterious beast in python, it does some really cool things though, like functional class creation (like in this instance), it's also the base class for metaclasses, which are also pretty interesting.

However, unittests really aren't the place to do interesting things, so I've replaced it with a class statement.